R/shared-methods.R

```diff
@@ -175,10 +116,65 @@ DataClass <- R6::R6Class(
     set_region_codes = function() {
     },
 
-    #' @description Initialize the country
-    #' @param ... Parameters passed to `initialise_dataclass`.
-    initialize = function(...) {
-      initialise_dataclass(self, ...)
+    #' @description initialize function used by all `Country` class objects.
+    #' Set up the country class with attributes set to input
+    #' parameters. Should only be called by a `Country` class object.
+    #' @param level A character string indicating the target administrative
+    #' level of the data with the default being "1". Currently supported
+    #' options are level 1 ("1) and level 2 ("2").
+    #' Use [get_available_datasets] for supported  options by dataset.
+    #' @param regions A character vector of target regions to be assigned to
+    #' the`target_regions` field if present.
+    #' @param totals Logical, defaults to FALSE. If TRUE, returns totalled
+    #'  data per region up to today's date. If FALSE, returns the full dataset
+    #'  stratified by date and region.
+    #' @param localise Logical, defaults to TRUE. Should region names be
+    #' localised.
+    #' @param verbose Logical, defaults to TRUE. Should verbose processing
+    #' @param steps Logical, defaults to FALSE. Should all processing and
+    #' cleaning steps be kept and output in a list.
+    #' @param get Logical, defaults to FALSE. Should the class `get` method be
+    #' called (this will download, clean, and process data at initialisation).
+    #' @export
+    initialize = function(level = "1", regions,
+                          totals = FALSE, localise = TRUE,
+                          verbose = TRUE, steps = FALSE, get = FALSE) {
+      if (any(self$supported_levels %in% level)) {
+        self$level <- level
+      } else {
+        stop(
+          level,
+          " is not a supported level check supported_levels for options"
```

Comment:
Did we not have tests of `initialise_dataclass` (whoops). Also need to remove it from the pkgdown site reference. Might be a good idea to check the built site locally as a way of debugging (`pkgdown::build_site()`)